libs/langgraph/langgraph/prebuilt/chat_agent_executor.py

```diff
@@ -426,7 +465,9 @@ class Agent,Tools otherClass
     else:
         tool_classes = tools
         tool_node = ToolNode(tool_classes)
```

Comment:
Unrelated to this but thoughts on getting the tool_classes from the tool node here too so we can let you pass in raw functions ?
```
        tool_classes = tools.tools_by_name.values()
```

Reply:
i think in `tools_by_name` it will also be a tool as we're calling `create_tool` under the hood https://github.com/langchain-ai/langgraph/blob/main/libs/langgraph/langgraph/prebuilt/tool_node.py#L95